### ISSUE TITLE
Brave doesn't try to connect to local IPv6 websites if we don't have IPv6 internet 

### ISSUE THREAD

**Issue by analotia** [brave/brave-browser#11874]
## Description 

I can't connect to a local IPv6 website which only has an AAAA record. Basically if the domain only has an AAAA record (and has no A record ), Brave never tries to get the AAAA record and connect to it.

Brave disables IPv6 connectivity for all domains even with local IPv6 websites that don't need to be routed over the IPv6 internet.   

This only impacts local IPv6-only domain names.  

I tried adding the website to `/etc/hosts` and I also tried the `--enable-ipv6 ` flag. Both didn't work

## Steps to Reproduce

   1.  Try to visit local IPv6 only website without IPv6 internet connectivity
   2.  Brave fails to load the page

## Actual result:
Brave doesn't try to get AAAA record and just fails with `ERR_NAME_NOT_RESOLVED`

## Expected result:
Brave should attempt to get AAAA record and try to connect to the IPv6 website regardless of whether we have IPv6 internet

## Desktop Brave version:

Brave | 1.14.84 Chromium: 85.0.4183.121 (Official Build) (64-bit)
-- | --
Revision | a81aa729a8e1fd413943a339393c82e7b8055ddc-refs/branch-heads/4183@{#1864}
OS | Linux

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
Yes.
- Can you reproduce this issue with the beta channel? 
Yes.
- Can you reproduce this issue with the nightly channel? 
Yes.

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
No.
- Does the issue resolve itself when disabling Brave Rewards? 
No.
- Is the issue reproducible on the latest version of Chrome? 
Yes. https://bugs.chromium.org/p/chromium/issues/detail?id=530482

**Comment by analotia** [comment]
Could you patch out the IPv6 connectivity check? This makes it impossible for me to switch from Firefox to Brave

**Comment by analotia** [comment]
Here is the code that does the IPv6 connectivity check: https://chromium.googlesource.com/chromium/src/+/master/net/dns/host_resolver_manager.cc#130

It also seems to send out connections to Google

**Comment by analotia** [comment]
https://github.com/Eloston/ungoogled-chromium/blob/master/patches/core/inox-patchset/0009-disable-google-ipv6-probes.patch  
https://github.com/Eloston/ungoogled-chromium/blob/master/patches/extra/ungoogled-chromium/add-ipv6-probing-option.patch

Here are the patches for this, seems like it's already done

**Comment by bsclifton** [comment]
cc: @fmarier @iefremov

**Comment by iefremov** [comment]
I guess it's not a problem to add these patches, just need to check with @brave/sec-team and @brave/patch-reviewers

**Comment by diracdeltas** [comment]
this was discussed in https://github.com/brave/brave-browser/issues/2324

https://github.com/Eloston/ungoogled-chromium/blob/master/patches/extra/ungoogled-chromium/add-ipv6-probing-option.patch seems ok to support in Brave AFAICT

**Comment by rany2** [comment]
@diracdeltas 

The issue I have with the IPv6 probes is that I need Brave to resolve IPv6-only intranet domains. I only have IPv4 internet connectivity with a private IPv6 address.  

#2324 is just for setting the probes to use something other than Google DNS AFAIK. Here I'm asking to get rid of the IPv6 probes altogether.

**Comment by diracdeltas** [comment]
that's correct, we discussed both options in #2324 but mostly the option to switch the DNS provider. would adding the command line flag to disable probing altogether (off by default) solve your use case?

**Comment by rany2** [comment]
@diracdeltas a command line flag to disable probing would be sufficient. I don't mind it not being default

**Comment by AbdoGramming** [comment]
It works with me when I wrap the ipv6 address within []
like [::1]

**Comment by rany2** [comment]
That's not the issue. This is about local IPv6 hostnames not being resolved.

**Comment by anthoeknee** [comment]
I second this happening. I am on the latest stable version of Mac OS as of January 30th of 2025.

**Comment by walmartshopper** [comment]
+1, ran into this today.  I don't have ipv6 internet connectivity, but I have a wireguard tunnel that routes some specific ipv6 addresses to some cloud-based web servers that only have AAAA records.  When I try to load a website on one of these servers in Brave, I get "This site can't be reached, DNS address could not be found.  DNS_PROBE_POSSIBLE"

I can load the website with no issues in Chromium and Firefox.

**Comment by danrask** [comment]
I’m experiencing the same issue with IPv6-only sites on our corporate intranet. `dig -t AAAA` and `ping6` work fine, other browsers have no problems as well — this only happens with Brave.

Currently, I've resolved this issue by adding the IPv6-only local sites directly to the `/etc/hosts`. Fortunately, we only have a handful of those; most are accessible via IPv4. Still, it's quite a hacky & cumbersome workaround.
